Makefile

```diff
@@ -69,14 +77,6 @@ docker-build:
 docker-push:
 	$(CONTAINER_CMD) push ${IMG}
 
-# find or download controller-gen
 # download controller-gen if necessary
-controller-gen:
-ifeq (, $(shell which controller-gen))
-	@{ \
-	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.2 ;\
-	}
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+$(CONTROLLER_GEN):
+	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.2
```

Comment:
Why have you renamed it from `controller-gen` to `$(CONTROLLER_GEN)`?
Now the user would need to specify the full path with make like `make /usr/bin/controller-gen` when he would need to install it, but that can change.

Reply:
Ok, updated. I didn't think it is necessary but I have created also a phony target `controller-gen`. It allows one to install `controller-gen` simply with `make controller-gen` if one wants. And hopefully it will also make the Makefile dependencies little bit easier to follow.